### PR TITLE
Add definitions for CodeMirror's runmode addon

### PR DESCRIPTION
**Pull request by Joev-**
Adds definitions for CodeMIrror's [runmode](http://codemirror.net/doc/manual.html#addon_runmode) addon

### CODE REVIEW

**Comment by vvakame** [comment]
@mihailik @basarat could you review this PR?

File: codemirror/codemirror-runmode.d.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+     *
+     * @param text The document to run through the highlighter.
+     * @param mode The mode to use (must be loaded as normal).
+     * @param output If this is a function, it will be called for each token with
+     *               two arguments, the token's text and the token's style class
+     *               (may be null for unstyled tokens). If it is a DOM node, the
+     *               tokens will be converted to span elements as in an editor,
+     *               and inserted into the node (through innerHTML).
+     */
+    function runMode(text : string, mode : any, output : any): void;
```
Comment (mihailik) [inline codemirror/codemirror-runmode.d.ts]: Maybe let's use the argument names [from the original?](https://github.com/codemirror/CodeMirror/blob/master/addon/runmode/runmode.js#L14)

```javascript
CodeMirror.runMode = function(string, modespec, callback, options) {
```

Can also peg `callback` argument to a union type: `callback: (HTMLElement | (text: string, style: string) => void)`

Optional last argument can also be added something like: `options?: { tabSize?: number; state?: any; }` — although that one is probably rarely used so can skip for now.

**Comment by Joev-** [comment]
I will get to making those changes when I am back in work next week.

**Comment by larshp** [comment]
hi, any updates on this PR? How do we proceed? @mihailik

**Comment by Joev-** [comment]
Is there anything else I needed to add or improve?

**Comment by basarat** [comment]
Thanks :rose:

**Pull request merged**